omnigibson/robots/behavior_robot.py

```diff
@@ -358,16 +359,116 @@ def base_footprint_link_name(self):
+        # If the simulator is playing, set the 6 base joints to achieve the desired pose of base_footprint link frame
+        if og.sim.is_playing() and self.initialized:
+            # Find the relative transformation from base_footprint_link ("base_footprint") frame to root_link
+            # ("base_footprint_x") frame. Assign it to the 6 1DoF joints that control the base.
+            # Note that the 6 1DoF joints are originated from the root_link ("base_footprint_x") frame.
+            if frame == "world" or frame == "scene":
+                # get the pose of the root link in the world/parent frame, invert to get the pose from root_link to world/scene frame
+                joint_pos, joint_orn = self.root_link.get_position_orientation(frame=frame)
+            else:
+                # getting root link pose in the parent frame is tricky, use PoseAPI to get the parent of the robot directly
+                joint_pos, joint_orn = PoseAPI.get_position_orientation(self.prim_path, frame="parent")
+            inv_joint_pos, inv_joint_orn = T.mat2pose(T.pose_inv(T.pose2mat((joint_pos, joint_orn))))
+
+            relative_pos, relative_orn = T.pose_transform(inv_joint_pos, inv_joint_orn, position, orientation)
+            relative_rpy = T.quat2euler(relative_orn)
+            self.joints["base_x_joint"].set_pos(relative_pos[0], drive=False)
+            self.joints["base_y_joint"].set_pos(relative_pos[1], drive=False)
+            self.joints["base_z_joint"].set_pos(relative_pos[2], drive=False)
+            self.joints["base_rx_joint"].set_pos(relative_rpy[0], drive=False)
+            self.joints["base_ry_joint"].set_pos(relative_rpy[1], drive=False)
+            self.joints["base_rz_joint"].set_pos(relative_rpy[2], drive=False)
+
+        # Else, set the pose of the robot frame, and then move the joint frame of the world_base_joint to match it
+        else:
+            # Call the super() method to move the robot frame first
+            super().set_position_orientation(position, orientation, frame=frame)
+
+            # convert the position and orientation to world frame
+            if frame == "scene":
+                if self.scene is None:
+                    raise ValueError("Cannot set pose relative to scene without a scene.")
+                else:
+                    position, orientation = T.mat2pose(self.scene.pose @ T.pose2mat((position, orientation)))
+            elif frame == "parent":
+
+                # get the parent prim path
+                parent_prim_path = "/".join(self.prim_path.split("/")[:-1])
+                parent_position, parent_orientation = PoseAPI.get_position_orientation(parent_prim_path)
+
+                # combine them to get the pose from root link to the world frame
+                position, orientation = T.pose_transform(parent_position, parent_orientation, position, orientation)
```

Comment:
There seems to be some weirdness here but in general I think the meaning of `local pose` is really murky for a robot. Leave this comment open for discussion on Friday.